src/extension/extension.ts

```diff
@@ -1,19 +1,19 @@
-import { EventEmitter } from 'events';
+import { EventEmitter } from 'node:events';
+import path from 'node:path';
 import { LanguageClient, SettingMonitor, ExecuteCommandRequest } from 'vscode-languageclient/node';
-import { workspace, commands, window } from 'vscode';
+import { workspace, commands, window, type ExtensionContext } from 'vscode';
 import { ApiEvent, PublicApi } from './types';
 import {
 	CommandId,
 	DidRegisterDocumentFormattingEditProviderNotificationParams,
 	Notification,
 } from '../server';
-import type vscode from 'vscode';
 
 /**
  * Activates the extension.
  */
-export function activate({ subscriptions }: vscode.ExtensionContext): PublicApi {
-	const serverPath = require.resolve('./start-server');
+export function activate({ subscriptions }: ExtensionContext): PublicApi {
+	const serverPath = path.join(__dirname, 'start-server.js');
```

Comment:
`__dirname` is a CommonJS thing. In ES context it shouldn't be used ideally (even if it's supported by some bundlers, compilers). Check https://stackoverflow.com/a/50052194

Reply:
Right. But for now, `__dirname` works like:
https://github.com/stylelint/vscode-stylelint/blob/5a40d977f94a49bf5cae24e963bc0964b3302a28/src/extension/start-server.ts#L11

We can switch the code base to ESM later. I'd like to keep this PR as small as possible.